_posts/2021-02-01-aws-config-compliance-automation.md

```diff
@@ -0,0 +1,95 @@
+
+Storing metadata around the process is helpful to, and any persistence store desired can be used. AWS DynamoDB lends itself very well to simple data storage needs, and a sample set of information stored in DynamoDB which could assist a process such as compliance automation might look something like:
+
+| Field      | Description |
+| ----------- | ----------- |
+| resourceId      | The AWS Config `resourceId`       |
+| ruleName   | The AWS Config rule name reporting non-compliance        |
+| accountId   | The AWS Account holding the resource        |
+| resourceStatus   | An internal tracking status for the integration        |
+| serviceNowItem   | Item from ServiceNow used to hold a single non-compliance finding (user-friendly number & `sys_id`)        |
+| serviceNowItem   | Request from ServiceNow, contains one-to-many `serviceNowItem`'s  (user-friendly number & `sys_id`)        |
+{: .table}
+
+This tracking information can then be utilized by the different steps in the Step Function to determine the adequate course of action (open ServiceNow item, close ServiceNow item, no action, etc).
+
+## All Together Now
+
+Creation of ServiceNow items is great, but only really starts to shine when it fits or enhances the normal way of confronting compliance issues for the organization. Normally, an organization will have "runbooks" or ways of addressing issues as they arise. An organization will also normally categorize compliance issues, according to their risk/impact to security posture. 
+
+All of this data is vital to ensuring that the compliance issues are addressed appropriately. For instance, an organization that deems a compliance rule as very `CRITICAL` will want to make sure that anything found to be in non-compliance with that rule is addressed as soon as possible. This might mean anything from separate alerting/notifications, to putting a hard deadline on action towards resolution (i.e. a `CRITICAL` finding might need to be addressed within the day, while something with a `LOW` criticality could be put off for a couple of weeks or more).
+
+Data in an organization's "runbook" might detail the normal course of remediation for resources found non-compliant with a given rule. This information is very helpful if a team is responsible for addressing findings within ServiceNow, and ServiceNow Item's description can list this as an added help to anyone that might be tasked with moving the resource back towards being in compliance.
+
+## Conclusion
+
+Overall, increases in automation generally assist in increasing an IT Operation's teams productivity. Tracking down compliance issues and manually remediating them is normallly very time consuming and leaves a lot of room for error. Compliance Automation can help increase visibility, reduce mean-time to resolution, contribute to a better overall security posture, and ensure compliance requirements are capable of being met with less overall work.
```

Comment:
"Overall, added automation generally assists in increasing IT Operation teams productivity."

"normally"